### ISSUE TITLE
Intelisense / compiler doesn't work in JSX for imported React components

### ISSUE THREAD

**Issue by niba** [Microsoft/TypeScript#10658]
TypeScript Version:  2.0-rc and 2.0-nightly

Create react component (stateless or normal)

```ts
interface DropdownProps {
  options: string[];
  onChange: (param: any) => void;
  option: string;
}

class Dropdown extends React.Component<DropdownProps, void> {
    render() {
        
        return (
             <select
                value={this.props.option}
                onChange={
                    (e) => { this.props.onChange(e.target.value); }
                }
            >
                {this.props.options.map(
                    (label, key) =>
                        <option value={key} key={key}>{label}</option>
                )}
            </select>
        )
    }
}
```

and try to use it in other component

```ts
import Dropdown from '../../components/Dropdown/Dropdown';

class Login extends React.Component<LoginPageProps, void> {
    render() {
       
        return (
            <div>
              <Dropdown  option={{}} />
  
            </div>
        )
    }
}

```

Intelisense doesnt show properties defined in Dropdown. You can pass any type and there are no errors.

I use vscode. In typescript 1.8 it was working.

**Comment by mhegazy** [comment]
This sample has a few issues, e.g. `Dropdown` is never exported, `LoginPageProps` is never imported, `React` is never imported, etc.. so i am assuming these are typos.

With that said, i do not see issues with this sample in VSCode today. here is screen cast of what i see:

![animation](https://cloud.githubusercontent.com/assets/8000722/18180250/e089b048-703a-11e6-8256-42da61b00b2d.gif)

**Comment by niba** [comment]
@mhegazy I didn't put entire code, only the important parts. Of course my components are exported and react and props are imported. 

On the bottom of your Code on the blue bar there is written `TypeScript React custom`. In my VSCode I have `TypeScript React 2.0.0`. Are you sure that you have 2.0 version? If yes why your code doesnt show the version as mine?

**Comment by mhegazy** [comment]
that is a nightly version, here is my version `2.1.0-dev.20160829`.

**Comment by niba** [comment]
@mhegazy and @RyanCavanaugh here is screen cast of my problem. I needed to restart VSCode after  changing version of Typescript. I recorded it on a different smaller project but the case is the same
![typescriptreactbug](https://cloud.githubusercontent.com/assets/2017454/18197364/eb71fb02-70f6-11e6-865e-6ac51f46ac6a.gif)

**Comment by mhegazy** [comment]
can you share a repro project i can try?

**Comment by niba** [comment]
Here is the project from my screen cast
https://gitlab.com/niba/air
`npm install` and `tsd install` to get it working.

One more thing. I created the minimal project, very similar to your from [comment](https://github.com/Microsoft/TypeScript/issues/10658#issuecomment-244176868) and the problem doesn't occur.

**Comment by niba** [comment]
@mhegazy What's more the code from first comment doesn't compile in Typescript 1.8 because key is type of number and property `value` in option expects string.

![typescriptreactbugoption](https://cloud.githubusercontent.com/assets/2017454/18347040/ae5021d2-75c3-11e6-9d3a-29dd3b029b95.png)

**Comment by mhegazy** [comment]
i am unable to get it to reproduce on https://gitlab.com/niba/air. seems to be working fine to me.

**Comment by niba** [comment]
Have you installed typescript@next?

**Comment by mhegazy** [comment]
yes

**Comment by niba** [comment]
Hmm, and what with this [comment](https://github.com/Microsoft/TypeScript/issues/10658#issuecomment-245563361). You should have had problems with compiling Dropdown code and I didn't see that on your screen cast

**Comment by mhegazy** [comment]
I see the issue now. it ended up being a file name casing issue. in index.tsx, you have an import to `./Main`, where as the file on disk is called `main.tsx`. this got the error checker confused. i should get a fix for this soon.

**Comment by niba** [comment]
You are right. It solved all problems in air project. Thanks!

Unfortunately it didn't help in my main project. I still don't have any intellisense in React components. I started removing file by file to discover which import breaks typescript and I ended up with one main file.
Is it possible that some typings break compiler?

Here is repo with what has left of project: https://github.com/niba/typescript_react_bug
One react file with a bunch of npm libraries and typings and it's enough to break typescript react intellisense (div is `any` type)
